### PR TITLE
Google cloud storage

### PR DESCRIPTION
**Pull request by howhelan**

### CODE REVIEW

**Comment by capt-haddock** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **howhelan**
You can assign the PR to them by writing `/assign @howhelan` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=pauwels-labs%2Fredact-crypto).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/pauwels-labs/redact-crypto/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by capt-haddock** [comment]
@howhelan: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-test | 6abc1cd72acf816657316c0f768132277ee6b895 | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/abs-redact-crypto-pr-6-unit-test-7tnjr) | `/test unit-test`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: src/entry.rs
Comment (ajpauwels) [inline src/entry.rs:126]: Don't we still need to check if the already filled OnceCell is a reference, and dereference it if so?

File: src/error.rs
Comment (ajpauwels) [inline src/error.rs:42]: Can we add a

// TODO: Figure out a better way to do this

comment to this for future notes? The general idea is I can do a general text search on the codebase for `todo` and it'll yield all the things we don't really have a better idea for or don't want to deal with right now and aren't that important but could have better solutions. This is also a standard Rust thing:
https://doc.rust-lang.org/std/macro.todo.html

File: src/storage.rs
Comment (ajpauwels) [inline src/storage.rs:28]: can we rename the module to just `google` or `gcs` for simplicity/lack of underscores sake?

**Review by ajpauwels — CHANGES_REQUESTED** [review]

File: src/error.rs
Comment (howhelan) [reply src/error.rs:42]: ignoring and putting up another PR with an actual fix for this

File: src/storage.rs
Comment (howhelan) [reply src/storage.rs]: renamed to gcs

**Pull request merged**
